File: src/3005_CountElementsWithMaximumFrequency/3005.go
func maxFrequencyElements(nums []int) int {
    freq := make(map[int]int)
    for _, n := range nums {
        freq[n]++
    }
    maxf := 0
    count := 1
    for _, v := range freq {
        if v > maxf {
            maxf = v
            count = 1
        } else if v == maxf {
            count++
        }
    }

    return count * maxf
}

File: src/3005_CountElementsWithMaximumFrequency/3005.rb
# @param {Integer[]} nums
# @return {Integer}
def max_frequency_elements(nums)
    freq = nums.tally.sort_by{|k,v| v}
    maxf = freq.last[1]
    return freq.filter_map{|arr| arr[1] if arr[1] == maxf}.sum
end